frontend_app_gallery/test/component/Carousel.spec.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { expect } from 'chai'
+import { shallow } from 'enzyme'
+import { Carousel as CarouselWithoutHOC } from '../../src/component/Carousel.jsx'
+import sinon from 'sinon'
+
+describe('<Carousel />', () => {
+  const onCarouselPositionChange = sinon.spy()
+
+  const props = {
+    onCarouselPositionChange: onCarouselPositionChange,
+    isWorkspaceRoot: false,
+    slides: [
+      {
+        src: 'testSrc1',
+        rotationAngle: 0
+      }
+    ]
+  }
+
+
+  const wrapper = shallow(<CarouselWithoutHOC {...props} t={tradKey => tradKey} />)
```

Comment:
**shallow** vs **mount**

**shallow** : Render only the component without his children components
Allows to write real unit test
**mount** : Render the component and his children components
Allows to test the component with his children, the only way to test `componentDidMount` et `componentDidUpdate`, higher cost in execution time

**Conclusion** : Always start to test a component with **shallow**, but if we need to test functions `componentDidMount` and `componentDidUpdate` or his children behavior, then **mount** is needed.

[Source](https://github.com/airbnb/enzyme/issues/465#issuecomment-227697726)

In this case I think there is no need of mount because we only need the props of Slider in just one test.

Reply:
I agree